request.js

```diff
@@ -778,6 +778,8 @@ Request.prototype.start = function () {
           var e = new Error('ESOCKETTIMEDOUT')
           e.code = 'ESOCKETTIMEDOUT'
           e.connect = false
+          e.host = self.host
+          e.port = self.port
```

Comment:
Ah no, url.parse returns port as a string. But we're setting it manually as a number in other places . 

We should change to always be a number or string in a future PR.